diff --git a/getTemplate.cgi b/getTemplate.cgi
index ff0db33..f90caf1 100755
--- a/getTemplate.cgi
+++ b/getTemplate.cgi
@@ -34,7 +34,7 @@ except:
 
 import CGI
 import string
-import regex
+import re
 import pg_db
 db = pg_db      # set up alias to minimize code changes
 
@@ -42,7 +42,7 @@ class getTemplateCGI (CGI.CGI):
 
   # CLASS VARIABLES
   pageElements = {} # holds element file -> file content mapping
-  re_parm = regex.compile ('\${\([^}]+.element\)}') # to find 'foo.element'
+  re_parm = re.compile ('\\${([^}]+.element)}') # to find 'foo.element'
   
 
   def resolve (self,
@@ -69,10 +69,14 @@ class getTemplateCGI (CGI.CGI):
     if steps == 0:
             raise error, 'Could not resolve parameter.'
     s = getTemplateCGI.pageElements[file]
-    while getTemplateCGI.re_parm.search (s) != -1:
-            start, stop = getTemplateCGI.re_parm.regs[0]
-            parm = getTemplateCGI.re_parm.group(1)
+
+    re_match = getTemplateCGI.re_parm.search (s)
+    while re_match:
+            start, stop = re_match.span()
+            parm = re_match.group(1)
             s = s[:start] + self.resolve(parm, steps-1) + s[stop:]
+            re_match = getTemplateCGI.re_parm.search (s)
+
     return s
 
   def getDbDate (self):
